lib/foreman_maintain.rb

```diff
@@ -139,5 +139,17 @@ def storage(label = :default)
     rescue StandardError => e
       logger.error "Invalid Storage label i.e #{label}. Error - #{e.message}"
     end
+
+    def perform_self_upgrade
+      puts 'Checking for new version of foreman-maintain...'
+      if ForemanMaintain.package_manager.update_available?('rubygem-foreman_maintain')
+        puts "\nUpdating rubygem-foreman_maintain package."
+        ForemanMaintain.package_manager.update('rubygem-foreman_maintain', :assumeyes => true)
+        puts "\nThe rubygem-foreman_maintain package successfully updated."\
+            "\nRe-run satellite-maintain or foreman-maintain with required options!"
```

Comment:
As offline discussion we decided to show both but after giving second thought it would be good to show only one at a time.
For upstream users, it will always show `foreman-maintain` but for downstream I think it should show only `satellite-maintain`

Reply:
@upadhyeammit, @jameerpathan111, what do you think?

Reply:
I am OK with either.

Reply:
Hello @kgaikwad I have added method to handle it as per upstream and downstream. Request to review, thank you !